### PR TITLE
FEAT-#2796: Add spreadsheet API

### LINKED ISSUE #2796: Add Spreadsheet API
Add the `modin.spreadsheet` API by integrating `modin-spreadsheet`.

Plan to expose 2 API:
`from_dataframe`: Creates spreadsheet from dataframe
`to_dataframe`:  Exports dataframe from spreadsheet

Conventional usage could look like:
`import modin.spreadsheet as mss`

### PR DESCRIPTION
**Pull request by richardlin047**
## What do these changes do?
- Add spreadsheet API and expose as `modin.spreadsheet`
- Add `spreadsheet` to setup.py extras so it can be installed using `pip install modin[spreadsheet]`
- Add jupyter notebook tutorial
- Add basic testing for spreadsheet API

## Changes still needed
- (Decided to leave) Cut down documentation on `from_dataframe`? 
- (Decided to leave) Move tutorial location?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2796 ? 
- [x] tests added and passing

### CODE REVIEW

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
We need to add the tests to the CI and add modin_spreadsheet to the development environment and test environment files.

File: modin/spreadsheet/test/test_general.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    data = get_test_data()
+    modin_df = pd.DataFrame(data)
+    pandas_df = pandas.DataFrame(data)
+
+    spreadsheet = mss.from_dataframe(modin_df)
+    modin_result = mss.to_dataframe(spreadsheet)
+
+    assert modin_result.equals(modin_df)
+
+    with pytest.raises(TypeError):
```
[the remark this answers is not in the document]
Comment (richardlin047) [reply modin/spreadsheet/test/test_general.py:55]: Oops, should be a `to_dataframe`

**Comment by richardlin047** [comment]
Tests have been added to CI under `ci.yml` and `push.yml`. Also added `modin-spreadsheet` to `requirements-dev.txt` and `environment-dev.yml`.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -582,3 +582,33 @@ jobs:
...
+  test-spreadsheet:
+    needs: [ lint-commit, lint-flake8, lint-black, test-api, test-headers ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8" ]
+        engine: ["ray", "dask"]
+    env:
+      MODIN_BACKEND: ${{matrix.engine}}
+      MODIN_EXPERIMENTAL: "True"
```
Comment (devin-petersohn) [inline .github/workflows/ci.yml:595]:
```suggestion
```

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
I think the experimental mode is throwing it off.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (richardlin047) [reply .github/workflows/ci.yml:595]: Should I also reduce the matrix? Adds 6 tests to the CI

File: modin/spreadsheet/test/test_general.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        "B": pd.Timestamp("20130102"),
+        "C": pd.Series(1, index=list(range(4)), dtype="float32"),
+        "D": np.array([3] * 4, dtype="int32"),
+        "E": pd.Categorical(["test", "train", "foo", "bar"]),
+        "F": ["foo", "bar", "buzz", "fox"],
+    }
+
+
+def test_from_dataframe():
+    data = get_test_data()
```
Comment (richardlin047) [inline modin/spreadsheet/test/test_general.py:34]: @devin-petersohn I get NotImplementedError on this line for the CI tests, should I not be using helper methods?

Comment (richardlin047) [reply modin/spreadsheet/test/test_general.py:34]: This is the error: https://github.com/modin-project/modin/pull/2797/checks?check_run_id=2024302005

Comment (devin-petersohn) [reply modin/spreadsheet/test/test_general.py:34]: I think you just need to remove experimental flag. See my review above.

Comment (richardlin047) [reply modin/spreadsheet/test/test_general.py:34]: Looks like it's still failing: https://github.com/modin-project/modin/pull/2797/checks?check_run_id=2024670302

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -582,3 +582,32 @@ jobs:
...
+
+  test-spreadsheet:
+    needs: [ lint-commit, lint-flake8, lint-black, test-api, test-headers ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8" ]
+        engine: ["ray", "dask"]
+    env:
+      MODIN_BACKEND: ${{matrix.engine}}
```
Comment (devin-petersohn) [inline .github/workflows/ci.yml:594]:
```suggestion
      MODIN_ENGINE: ${{matrix.engine}}
```

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -304,3 +304,32 @@ jobs:
...
+
+  test-spreadsheet:
+    needs: [ lint-commit, lint-flake8, lint-black, test-api, test-headers ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.6", "3.7", "3.8" ]
+        engine: ["ray", "dask"]
+    env:
+      MODIN_BACKEND: ${{matrix.engine}}
```
Comment (devin-petersohn) [inline .github/workflows/push.yml:316]:
```suggestion
      MODIN_ENGINE: ${{matrix.engine}}
```

**Review by devin-petersohn — CHANGES_REQUESTED** [review]

**Review by devin-petersohn — APPROVED** [review]
Thanks @richardlin047 LGTM!

**Comment by rootsmusic** [comment]
@richardlin047 See #6416 about modin.spreadsheet.

**Pull request merged**